### ISSUE TITLE
byte_conversion and bytscl fail on aarch64, ppc64le, s390x

### ISSUE THREAD

**Issue by opoplawski** [gnudatalanguage/gdl#1079]
e.g. for s390x:
```
      Start  64: test_byte_conversion.pro
64: Test command: /builddir/build/BUILD/gdl-1.0.0/build/src/gdl "-quiet" "-e" "test_byte_conversion"
64: Environment variables: 
64:  LC_COLLATE=C
64:  GDL_PATH=/builddir/build/BUILD/gdl-1.0.0/testsuite/:/builddir/build/BUILD/gdl-1.0.0/src/pro/
64:  GDL_STARTUP=
64:  IDL_STARTUP=
64: Test timeout computed to be: 3600
64: % Compiled module: TEST_BYTE_CONVERSION.
64: % Compiled module: BANNER_FOR_TESTSUITE.
64: % Compiled module: GDL_IDL_FL.
64: % TEST_BYTE_BUG_586:   NO errors encountered during TEST_BYTE_BUG_586 tests  
64: % Compiled module: ERRORS_CUMUL.
64: % TEST_BYTE_BASIC_32768: 
64:   NO errors encountered during TEST_BYTE_BASIC_32768 tests  
64: % Compiled module: ERRORS_ADD.
64: % TEST_BYTE_BASIC_32768: Error on operation : input as Long for :       -32765
64: % TEST_BYTE_BASIC_32768: Error on operation : input as Long for :       -32766
64: % TEST_BYTE_BASIC_32768: Error on operation : input as Long for :       -32767
64: % TEST_BYTE_BASIC_32768: Error on operation : input as Long for :       -32769
64: % TEST_BYTE_BASIC_32768: Error on operation : input as Long for :       -32770
64: % TEST_BYTE_BASIC_32768: 
64: =============================================================
64: % TEST_BYTE_BASIC_32768: 
64: =                                                           =
64: % TEST_BYTE_BASIC_32768: 
64: =  5 errors encountered during TEST_BYTE_BASIC_32768 tests  =
64: % TEST_BYTE_BASIC_32768: 
64: =                                                           =
64: % TEST_BYTE_BASIC_32768: 
64: =============================================================
64: % TEST_BYTE_BASIC_32768: ==============================================
64: % TEST_BYTE_BASIC_32768: =                                            =
64: % TEST_BYTE_BASIC_32768: =  For details, run it again with /DEBUG !!  =
64: % TEST_BYTE_BASIC_32768: =                                            =
64: % TEST_BYTE_BASIC_32768: ==============================================
64: % TEST_BYTE_STRING2BYTE: 
64:   NO errors encountered during TEST_BYTE_STRING2BYTE tests  
64: 2017-03-11 : First run should be OK in GDL
64: % TEST_BYTE_WITH_LOOP: 
64:   NO errors encountered during TEST_BYTE_WITH_LOOP tests  
64: 2017-03-11 : Second run should be BAD in GDL
64: % TEST_BYTE_WITH_LOOP: 
64:   NO errors encountered during TEST_BYTE_WITH_LOOP tests  
64: % TEST_BYTE_CONVERSION: 
64: ============================================================
64: % TEST_BYTE_CONVERSION: 
64: =                                                          =
64: % TEST_BYTE_CONVERSION: 
64: =  5 errors encountered during TEST_BYTE_CONVERSION tests  =
64: % TEST_BYTE_CONVERSION: 
64: =                                                          =
64: % TEST_BYTE_CONVERSION: 
64: ============================================================
 2/10 Test  #64: test_byte_conversion.pro .........***Failed    0.16 sec
test 65
      Start  65: test_bytscl.pro
65: Test command: /builddir/build/BUILD/gdl-1.0.0/build/src/gdl "-quiet" "-e" "test_bytscl"
65: Environment variables: 
65:  LC_COLLATE=C
65:  GDL_PATH=/builddir/build/BUILD/gdl-1.0.0/testsuite/:/builddir/build/BUILD/gdl-1.0.0/src/pro/
65:  GDL_STARTUP=
65:  IDL_STARTUP=
65: Test timeout computed to be: 3600
65: % Compiled module: TEST_BYTSCL.
65: 
65: PLEASE contribute to add tests on MIN, MAX, TOP keywords
65: 
65: % Compiled module: DIST.
65: % Compiled module: ERRORS_ADD.
65: % TEST_BYTSCL_TOP: Error on operation : Pb with top=     -128, min/max :        0,        0
65: % Compiled module: BANNER_FOR_TESTSUITE.
65: % Compiled module: GDL_IDL_FL.
65: % TEST_BYTSCL_RAMPS: =========================================================
65: % TEST_BYTSCL_RAMPS: =                                                       =
65: % TEST_BYTSCL_RAMPS: =  1 errors encountered during TEST_BYTSCL_RAMPS tests  =
65: % TEST_BYTSCL_RAMPS: =                                                       =
65: % TEST_BYTSCL_RAMPS: =========================================================
65: % Compiled module: ERRORS_CUMUL.
65: % TEST_BYTSCL_RAMPS:   NO errors encountered during TEST_BYTSCL_RAMPS tests  
65: % TEST_BYTSCL_RAMPS_NAN: 
65:   NO errors encountered during TEST_BYTSCL_RAMPS_NAN tests  
65: % TEST_BYTSCL_IDL_PROBLEM: 
65:   NO errors encountered during TEST_BYTSCL_IDL_PROBLEM tests  
65: % TEST_BYTSCL: ===================================================
65: % TEST_BYTSCL: =                                                 =
65: % TEST_BYTSCL: =  1 errors encountered during TEST_BYTSCL tests  =
65: % TEST_BYTSCL: =                                                 =
65: % TEST_BYTSCL: ===================================================
 3/10 Test  #65: test_bytscl.pro ..................***Failed    0.18 sec
```

**Comment by ShamrockLee** [comment]
Same issue here on the aarch64-linux.

**Comment by GillesDuvert** [comment]
replaced by #1227

**Comment by GillesDuvert** [comment]
finally closed by #1949 
It would have been useful at that time to hear about the different instruction set for AARCH64 that did not convert floating-point values to unsigned integers the same as on intel machines...

**Comment by opoplawski** [comment]
It does look good now.
